scripts/zones/Aht_Urhgan_Whitegate/npcs/Kaduru-Haiduru.lua

```diff
@@ -0,0 +1,125 @@
+-----------------------------------
+-- Area: Aht Urhgan Whitegate
+-- NPC: Kaduru-Haiduru
+-- Teleport NPC
+-----------------------------------
+
+-----------------------------------
+-- onTrigger
+-----------------------------------
+function onTrigger(player, npc)
+    local caughtUsingShihuDanhuDate = player:getVar("Kaduru_ShihuDanhu_date");
+    local shihuDanhuDate = player:getVar("ShihuDanhu_TP_date");
+    local timesUsed = player:getVar("Kaduru_TimesUsed");
+    local currentDate = os.date("%j");
+
+    if ((canUse_KaduruHaiduru_Service(player) == 1)) then
+        player:startEvent(0x0097, 0, 0, timesUsed, 0, 0, 0, 0, 0, 0);
+    else
+        if (caughtUsingShihuDanhuDate == 0) then
+            player:setVar("Kaduru_ShihuDanhu_date", os.date("%j"));
+            player:setVar("Kaduru_TimesUsed", 0);
+        end
+        player:startEvent(0x0099, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+    end
+end
+
+function onTrade(player, npc, trade)
+    local caughtUsingShihuDanhuDate = player:getVar("Kaduru_ShihuDanhu_date");
+    local shihuDanhuDate = player:getVar("ShihuDanhu_TP_date");
+    local timesUsed = player:getVar("Kaduru_TimesUsed");
+
+    if ((canUse_KaduruHaiduru_Service(player) == 1) and (timesUsed == 3)) then
```

Comment:
`if canUse_KaduruHaiduru_Service(player) == 1 and timesUsed == 3 then`
Or if you take my true/false suggestion:
`if canUse_KaduruHaiduru_Service(player) and timesUsed == 3 then`

you should get the idea now without burying you in comments :smile:  you don't need the extra parens

Reply:
I see nvm on above comment.

Reply:
actually double checking now because that was a good question